src/ui/inspector/Location.vue

```diff
@@ -44,45 +46,31 @@ export default {
         this.openmct.selection.off('change', this.updateSelection);
     },
     methods: {
-        getObjectPath(domainObject) {
-            let objectPath = [],
-                context = domainObject.getCapability('context');
-
-            if (context) {
-                objectPath = context.getPath();
-            }
-
-            return objectPath;
-        },
-        itemDecorator(d) {
-            let domainObject = d.useCapability('adapter'),
-                key = this.openmct.objects.makeKeyString(domainObject.identifier),
-                childObjectPath = this.getObjectPath(d)
-                    .slice(1)
-                    .reverse()
-                    .map((dd) => dd.useCapability('adapter'));
-
-            return {
-                domainObject,
-                key,
-                objectPath: childObjectPath 
-            }
+        makeParentObjects() {
+            this.originalPath.forEach((pathObject) => {
+                this.openmct.objects.getOriginalPath(pathObject.key)
```

Comment:
Why do we need to get the original path for each object in the chain? Couldn't we just get it once for the child and slice() for each parent?